### ISSUE TITLE
ERROR: Coult not initialize the database

### ISSUE THREAD

**Issue by jeroenpraat** [rev138/pod_feeder#19]
I installed all missing Perl modules using Cspan and I installed Sqlite3.

But I receive the following error:
`DBD::SQLite::db prepare failed: no such table: feeds at ./pod_feeder.pl line 648.
ERROR: Coult not initialize the database: DBD::SQLite::db prepare failed: no such table: feeds at ./pod_feeder.pl line 648.`

After running:
`./pod_feeder.pl --feed-id myfeed --feed-url https://xxxxx --pod-url https://xxxxx --username xxxxx --password xxxxx --no-branding --timeout 4`

The file 'pod_feeder.db' is 0 bytes and it doesn't matter when I remove it. The script recreated it with 0 bytes. 

What can be the problem? Running on Debian Jessie and this not the first Perl script I run on there.

**Comment by elektrollart** [comment]
Same here on Arch Linux

**Comment by elektrollart** [comment]
I createt a db on my own with sqlite:

`CREATE TABLE feeds(guid VARCHAR(255) PRIMARY KEY,feed_id VARCHAR(127),title VARCHAR(255),link VARCHAR(255),image VARCHAR(255),image_title VARCHAR(255),hashtags VARCHAR(255),timestamp INTEGER(10),posted INTEGER(1),body VARCHAR(10000))`

**Comment by hknet** [comment]
you can solve it like this:

-----
diff -Naur pod_feeder.pl~ pod_feeder.pl                                                                                 
--- pod_feeder.pl~      2017-09-29 02:31:06.000000000 +0200
+++ pod_feeder.pl       2017-09-29 04:03:28.000000000 +0200
@@ -629,8 +629,8 @@
 sub init_database {
         my ( $db_file ) = @_;
 
-        my $dbh = connect_to_db( $db_file );
         unless( -e $db_file ){
+        my $dbh = connect_to_db( $db_file );
                 my $sth = $dbh->prepare(
                         'CREATE TABLE feeds(guid VARCHAR(255) PRIMARY KEY,feed_id VARCHAR(127),title VARCHAR(255),link VARCHAR(255),image VARCHAR(255),image_title VARCHAR(255),hashtags VARCHAR(255),timestamp INTEGER(10),posted INTEGER(1),body VARCHAR(10000))'
                 ) or die "Can't prepare statement: $DBI::errstr";
@@ -639,6 +639,7 @@
                 $dbh->disconnect();
         }
         else {
+                my $dbh = connect_to_db( $db_file );
                 my $sth = $dbh->column_info(undef, undef, 'feeds', undef);
                 my $body_exists = 0;
                 while( my( $tcat, $tscheme, $tname, $column_name ) = $sth->fetchrow_array() ) {
-----

the issue is as soon as you connect to the db the file is created in other words the "in case there is no db"-path is never reached as the file exists and then the alter table never works out.

**Comment by victorhck** [comment]
same issue here in openSUSE Tumbleweed running in a Raspberry Pi.

I installed all the dependencies. but after running:

 perl pod_feeder.pl --feed-id myfeed --feed-url https://victorhckinthefreeworld.com/feed/ --pod-url https://my_cute_pod --username victorhck --password my_cute_password --category-tags --title-tags

I get:

 BD::SQLite::db prepare failed: no such table: feeds at ./pod_feeder.pl line 650.
ERROR: Coult not initialize the database: DBD::SQLite::db prepare failed: no such table: feeds at ./pod_feeder.pl line 650.

I tried installing sqlite3 and create feeds.db by myself. NOK

any hint about that?
